L1Trigger/L1TTrackMatch/plugins/L1TrackJetProducer.cc

```diff
@@ -142,509 +143,237 @@ L1TrackJetProducer::~L1TrackJetProducer() {}
-      delete[] L1clusters[phibin];
-
-    // Now merge clusters, if necessary
-    for (int m = 0; m < nclust - 1; ++m) {
-      for (int n = m + 1; n < nclust; ++n) {
-        if (L2cluster[n].eta == L2cluster[m].eta && (std::abs(L2cluster[n].phi - L2cluster[m].phi) < 1.5 * phiStep_ ||
-                                                     std::abs(L2cluster[n].phi - L2cluster[m].phi) > 6.0)) {
-          if (L2cluster[n].pTtot > L2cluster[m].pTtot) {
-            L2cluster[m].phi = L2cluster[n].phi;
-          }
-          L2cluster[m].pTtot += L2cluster[n].pTtot;
-          L2cluster[m].numtracks += L2cluster[n].numtracks;
-          L2cluster[m].numtdtrks += L2cluster[n].numtdtrks;
-          for (unsigned int itrk = 0; itrk < L2cluster[n].trackidx.size(); itrk++)
-            L2cluster[m].trackidx.push_back(L2cluster[n].trackidx[itrk]);
-
-          for (int m1 = n; m1 < nclust - 1; ++m1) {
-            L2cluster[m1] = L2cluster[m1 + 1];
-          }
-          nclust--;
-          m = -1;
-          break;  //?????
-        }         // end if clusters neighbor in eta
-      }
-    }  // end for (m) loop
-
-    // sum up all pTs in this zbin to find ht
-    // Require jet to have at least 2 tracks with a summed eT>50 GeV, or 3 tracks with summed eT>100 GeV
-    // in order to count toward HT
-    float ht = 0;
-    for (int k = 0; k < nclust; ++k) {
-      if (L2cluster[k].pTtot > minJetEtLowPt_ && L2cluster[k].numtracks < lowpTJetMinTrackMultiplicity_)
+       }
+       for (int i = 0; i < phiBins_; ++i) {
+         for (int j = 0; j < etaBins_; ++j) {
+           float eta_min=epbins[i][j].eta - etaStep_ / 2.0; //eta min
+           float eta_max=epbins[i][j].eta + etaStep_ / 2.0; //eta max
+           float phi_min=epbins[i][j].phi - phiStep_ / 2.0; //phi min
+           float phi_max=epbins[i][j].phi + phiStep_ / 2.0; //phi max         
+
+           if ( ( trketa < eta_min ) || ( trketa > eta_max ) ||
+              ( trkphi < phi_min ) || ( trkphi > phi_max )   )
+              continue;
+
+           if (trkpt < trkPtMax_)
+             epbins[i][j].pTtot += trkpt;
+           else
+             epbins[i][j].pTtot += trkPtMax_;
+           epbins[i][j].numtdtrks += tdtrk_[k];
+           epbins[i][j].trackidx.push_back(k);
+           ++epbins[i][j].numtracks;
+        } // for each phibin
+      } // for each phibin
+   } //end loop over tracks
+
+   // first layer clustering - in eta for all phi bins
+   for (int phibin = 0; phibin < phiBins_; ++phibin) {
+     L1clusters.push_back( L1_clustering(epbins[phibin],etaBins_,etaStep_) );
+   }
+
+   //second layer clustering in phi for all eta clusters  
+   L2clusters= L2_clustering(L1clusters, phiBins_, phiStep_, etaStep_);
+
+ // sum all cluster pTs in this zbin to find max
+   float sum_pt = 0;
+   for (unsigned int k = 0; k < L2clusters.size(); ++k) {
+     if (L2clusters[k].pTtot > 50 && L2clusters[k].numtracks < lowpTJetMinTrackMultiplicity_)
```

Comment:
Define the magic number 50 at the beginning of the code

Reply:
Sure I will make it configurable. Thanks for catching it